assets/wizards/popups/views/campaigns/index.js

```diff
@@ -0,0 +1,331 @@
+									'newspack-campaigns__campaign-group__select-control-group-item',
+									+selectValue.key === +id && 'is-selected'
+								),
+							} ) ),
+							...( archivedCampaigns.length
+								? [
+										{
+											key: 'header-archived-campaigns',
+											name: __( 'Archived Campaigns', 'newspack' ),
+											className: 'is-header',
+										},
+								  ]
+								: [] ),
+							...archivedCampaigns.map( ( { term_id: id, name } ) => ( {
+								key: id,
+								name,
+								className: classnames(
+									'newspack-campaigns__campaign-group__select-control-group-item',
+									+selectValue.key === +id && 'is-selected'
+								),
+							} ) ),
+						] }
+						onChange={ ( { selectedItem: { key } } ) =>
+							'active' === key
+								? history.push( '/campaigns' )
+								: history.push( `/campaigns/${ key }` )
+						}
+						value={ selectValue }
+						hideLabelFromVision={ true }
+					/>
+					{ campaignData && (
+						<div className="newspack-campaigns__campaign-group__filter-group-actions__button">
+							<Button
+								isQuaternary
+								isSmall
+								className={ popoverVisible && 'popover-active' }
+								onClick={ () => setPopoverVisible( ! popoverVisible ) }
+								icon={ moreVertical }
+								label={ __( 'Actions', 'newspack' ) }
+							/>
+							{ popoverVisible && (
+								<CampaignManagementPopover
+									dismiss={ () => setPopoverVisible( false ) }
+									isArchive={ campaignData && 'archive' === campaignData.status }
+									onActivate={ () => manageCampaignGroup( prompts ) }
+									onArchive={ () => archiveCampaignGroup( campaignId, true ) }
+									onDeactivate={ () => manageCampaignGroup( prompts, 'DELETE' ) }
+									onDelete={ () => deleteCampaignGroup( campaignId ) }
+									onDuplicate={ () => {
+										setModalVisible( true );
+										setCampaignName( '' );
+										setModalType( MODAL_TYPE_DUPLICATE );
+									} }
+									onRename={ () => {
+										setCampaignName( campaignData.name );
+										setModalVisible( true );
+										setModalType( MODAL_TYPE_RENAME );
+									} }
+									onUnarchive={ () => archiveCampaignGroup( campaignId, false ) }
+									hasUnpublished={ prompts.some( ( { status } ) => 'publish' !== status ) }
+									hasPublished={ prompts.some( ( { status } ) => 'publish' === status ) }
+								/>
+							) }
+						</div>
+					) }
+				</div>
+				<div className="newspack-campaigns__campaign-group__add-new-button">
+					<Button
+						isPrimary
+						isSmall
+						onClick={ () => {
+							setModalVisible( ! modalVisible );
+							setCampaignName( '' );
+							setModalType( MODAL_TYPE_NEW );
+						} }
+					>
+						{ __( 'Add New Campaign', 'newspack' ) }
+					</Button>
+					{ modalVisible && (
+						<Modal
+							title={ modalTitle( modalType ) }
+							isDismissible={ false }
+							className="newspack-campaigns__campaign-group__add-new-button__modal"
+						>
+							<TextControl
```

Comment:
Would love for the `TextControl` to be focused as soon as this modal renders.

Reply:
This should handle the focus - https://github.com/Automattic/newspack-plugin/pull/833/commits/d03778500596268316c29a625e56ee9a183d8c4d

Reply:
Exactly what I wanted! Looks like it made the linter sad, though.

Reply:
Yup, linter is happy now: https://github.com/Automattic/newspack-plugin/pull/833/commits/0993b5a0b0999aeb652d65ee94f404d0f42b1943